components/create_team/components/display_name/display_name.jsx

```diff
@@ -11,20 +11,31 @@ import Constants from 'utils/constants.jsx';
 import {cleanUpUrlable} from 'utils/url.jsx';
 import logoImage from 'images/logo.png';
 
-export default class TeamSignupDisplayNamePage extends React.Component {
+export default class TeamSignupDisplayNamePage extends React.PureComponent {
+    static propTypes = {
+
+        /*
+         * Object containing team's display_name and name
+         */
+        state: PropTypes.object,
+
+        /*
+         * Function that updates parent component with state props
+         */
+        updateParent: PropTypes.func
+    }
+
     constructor(props) {
         super(props);
 
-        this.submitNext = this.submitNext.bind(this);
-
         this.state = {};
     }
 
     componentDidMount() {
         trackEvent('signup', 'signup_team_01_name');
     }
 
-    submitNext(e) {
+    submitNext = (e) => {
         e.preventDefault();
```

Comment:
Maybe good to modify in this PR.
I just notice that [below](https://github.com/mattermost/mattermost-webapp/pull/814/files#diff-20512d825303d95e16b3273071520678R64), we're modifying the value of `props.state`. Instead, I suggest to create `newState` and pass it to `props.updateParent`. Could be like:
```
let newState = {...this.props.state};
...
this.props.updateParent(newState);
```

Same goes [here at team_url.jsx](https://github.com/mattermost/mattermost-webapp/pull/814/files#diff-baac3881cb8c3e3a0869003798eee39dR60)